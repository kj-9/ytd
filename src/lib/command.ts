import * as log from 'std/log/mod.ts';
import logger from './logger.ts';
import { playlistItems } from './youtube.ts';
import config from '../../deno.json' with { type: 'json' };
import { Command } from 'cliffy/command/mod.ts';

const { name, version, description } = config;

async function main(options: any, args: string[]) {
  logger().debug('Fetching playlist items...');

  const res = await playlistItems(
    String(args[0]),
    String(args[1]),
  );
  console.log(res);
}

export default async function command() {
  await new Command()
    .name(name)
    .version(version)
    .description(description)
    .option('-d, --debug', 'Enable debug output.')
    .arguments('<playlistId:string> <apiKey:string>')
    .action(async (options, ...args) => {
      log.setup({
        handlers: {
          console: new log.ConsoleHandler(options.debug ? 'DEBUG' : 'INFO'),
        },

        loggers: {
          'ytd': {
            level: 'DEBUG',
            handlers: ['console'],
          },
        },
      });

      await main(options, args);
    })
    .parse(Deno.args);
}
