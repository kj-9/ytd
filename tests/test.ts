import { snapshotTest } from 'cliffy/testing/mod.ts';
import command from '../src/lib/command.ts';

await snapshotTest({
  name: 'should log help',
  meta: import.meta,
  args: ['--help'],
  async fn() {
    await command();
  },
});
