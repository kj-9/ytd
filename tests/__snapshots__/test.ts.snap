export const snapshot = {};

snapshot[`should log help 1`] = `
stdout:
"
Usage:   ytd <playlistId> <apiKey>
Version: 0.0.0-a4                 

Description:

  Youtube Data API CLI and library for Deno

Options:

  -h, --help     - Show this help.                            
  -V, --version  - Show the version number for this program.  
  -d, --debug    - Enable debug output.                       

"
stderr:
""
`;
